components/script/dom/element.rs

```diff
@@ -692,6 +693,29 @@ impl LayoutElementHelpers for LayoutJS<Element> {
         }
     }
 
+    #[allow(unsafe_code)]
+    fn get_lang_for_layout(&self) -> String {
+        unsafe {
+            let mut current_node = Some(self.upcast::<Node>());
+            while let Some(node) = current_node {
+                current_node = node.parent_node_ref();
+                match node.downcast::<Element>().map(|el| el.unsafe_get()) {
+                    Some(elem) => {
+                        if let Some(attr) = (*elem).get_attr_val_for_layout(&ns!(xml), &local_name!("lang")) {
+                            return attr.to_owned();
+                        } else if let Some(attr) = (*elem).get_attr_val_for_layout(&ns!(), &local_name!("lang")) {
```

Comment:
nit: (optional), I think using the pattern:

```rust
if foo {
    return;
}
if bar {
    return
}
```

Is more easy to read forward than:

```rust
if foo {
    return;
} else if bar {
    return;
}
```

But your call, I won't insist on this.